content/src/controller.js

```diff
@@ -734,14 +734,19 @@ function saveAction(forceOverwrite, loginPrompt, doneCallback) {
   }
   var doc = view.getPaneEditorData(paneatpos('left'));
   var filename = modelatpos('left').filename;
+  var thumbnailDataUrl = '';
   if (!doc) {
     // There is no editor on the left (or it is misbehaving) - do nothing.
     console.log("Nothing to save.");
     return;
+  } else if (doc.data !== '') { // If program is not empty, generate thumbnail.
+    thumbnailDataUrl = generateThumbnailDataUrl();
   }
   // Remember meta in a cookie.
   saveDefaultMeta(doc.meta);
-  var newdata = $.extend({}, modelatpos('left').data, doc);
+  var newdata = $.extend({
```

Comment:
You've added this to "saveAction" - can you also add thumbnailing to "shareAction", above in controller.js?  The idea is that when users share their programs to share.pencilcode.net, it should also add a thumbnail.